### ISSUE TITLE
2024.01-alpha2 stdout is swamped by a warning increasing its size 1000 fold

### ISSUE THREAD

**Issue by nikizadehgfdl** [NOAA-GFDL/FMS#1430]
**Describe the bug**
In the tag 2024.01-alpha2 stdout is swamped by the following warning for regression tests
```
 WARNING from PE  1807: data_override: current time outside bounds, use [previous]:current:[next] files in data_table
```

**To Reproduce**
Run ESM4.2 regression test

**Expected behavior**
Should be no such warning or at most a NOTE or FATAL if this is a problem

**Additional context**
The code is newly  added  to main branch by PR #1408 . There are many lines of code with this warning

https://github.com/NOAA-GFDL/FMS/pull/1408/files#diff-482d9c64ab36090ff35a1709f8ad19050a7d6817855d6a2d897096bfc51e0f36R845

**Comment by nikizadehgfdl** [comment]
Anyone knows what is wrong with this logic and why it is tripped in seemingly fine regression tests?
```
 if ((time<first_record) .or. (time>last_record)) then
         call mpp_error(WARNING, &
              'data_override: current time outside bounds, use [previous]:current:[next] files in data_table')
  endif
```

**Comment by nikizadehgfdl** [comment]
@raphaeldussin we should fix this. This check was not in the original code, perhaps it should not be in the new code either?

**Comment by nikizadehgfdl** [comment]
I got curious and put print statements to see what's going on. Indeed the logic above is tripped thousands of times with values like
```
time:                day=0, sec=0, ticks=0 
first_record:   day=15, sec=0, ticks=0
last_record:    day=350, sec=0, ticks=0
```
or
```
time:                day=0, sec=7200, ticks=0
first_record:   day=693499, sec=64800, ticks=0 
last_record:    day=693499, sec=64800, ticks=0
```

I don't know what is going on. But given that models run with these warnings, we should remove the if blocks and let it be like it was before. Ignorance is bliss.

**Comment by raphaeldussin** [comment]
@nikizadehgfdl This sounds dangerous. Removing the `if` blocks would not let the user know what is the problem with the time axis and how to fix it. This logic should work. Do we allow the current time to be outside of the time bounds or are we getting bad time bounds in the tests ?

**Comment by nikizadehgfdl** [comment]
In ESM4 case most of these warnings are because the data file has only 1 time level at 12/31/1899, (I guess that's where day=693499 comes from above). I think time_interp handles constant in time source data.

**Comment by rem1776** [comment]
fixed with #1431
